wp-content/themes/citylimits/inc/widgets/class-citylimits-special-projects-widget.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/**
+ * Register the widget
+ */
+add_action( 'widgets_init', function() {
+	register_widget( 'citylimits_special_projects_widget' );
+});
+
+/*
+ * List all of the terms in a custom taxonomy
+ */
+class citylimits_special_projects_widget extends WP_Widget {
+
+	function __construct() {
+		$widget_ops = array(
+			'classname' 	=> 'citylimits-special-projects',
+			'description' 	=> __('Display special projects', 'citylimits')
+		);
+		parent::__construct( 'citylimits-special-projects-widget', __('City Limits Special Projects', 'citylimits'), $widget_ops);
+	}
+
+	function widget( $args, $instance ) {
+		global $post;
+		// Preserve global $post
+		$preserve = $post;
+
+		// instance: num, series (id), title, heading
+
+		//get the posts
+		$series_1_posts = citylimits_get_series_posts( $instance['series_1'], $instance['num'], 'featured, DESC' );
+		$series_2_posts = citylimits_get_series_posts( $instance['series_2'], $instance['num'], 'featured, DESC' );
+		$series_3_posts = citylimits_get_series_posts( $instance['series_3'], $instance['num'], 'featured, DESC' );
+
+		$series_arr = [$series_1_posts, $series_2_posts, $series_3_posts];
+
+		if ( empty( $series_1_posts ) && empty( $series_2_posts ) && empty( $series_3_posts ) ) return; //output nothing if no posts found
+
+		$widget_title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'], $instance, $this->id_base );
+
+		echo $args['before_widget'];
+
+		echo '<div class="citylimits-special-projects-container">';
+
+		if ( ! empty( $widget_title ) ) echo $args['before_title'] . $widget_title . $args['after_title'];
+
+		$series_counter = 0;
+
+		echo '<div class="citylimits-special-projects-inner-container">';
+
+		foreach( $series_arr as $series ) {
+
+			$series_counter++;
+
+			if( $series->have_posts() ) {
+				
+				echo '<div class="citylimits-special-project">';
+
+				$thumbnail = '';
+				$series_id = $instance['series_'.$series_counter];
+				$title_link = get_term_link( (int) $instance['series_'.$series_counter], 'series' );
+				$term = get_term( $instance['series_'.$series_counter], 'series' );
+				$title = apply_filters( 'widget_title', $term->name, $instance, $this->id_base );
+				$excerpt = isset( $instance['excerpt_display'] ) ? $instance['excerpt_display'] : 'num_sentences';
+				
+				// get the term meta post so we can grab the featured image
+				$term_meta_post = largo_get_term_meta_post( 'series', $series_id );
+
+				if( largo_has_featured_media( $term_meta_post ) ){
+
+					$thumbnail = largo_get_featured_media( $term_meta_post );
+					$thumbnail = $thumbnail['attachment_data']['sizes']['large']['url'];
```

Comment:
Not all images have a 'large' size; would it be easier to use the get_thumbnail function against `$term_meta_post`?

The left-hand series here has this image:

![Screen Shot 2019-10-30 at 23 10 15 ](https://user-images.githubusercontent.com/1754187/67916177-44737300-fb6b-11e9-8d6a-fd26498a1dbe.png)
![agenda-19-v2](https://user-images.githubusercontent.com/1754187/67916188-4b9a8100-fb6b-11e9-8c23-ac246e4551d6.png)

Reply:
The problem is that the image hasn't a "large" size. For the Agenda 2019 image above, here's what `error_log(var_export( array_keys( $thumbnail['attachment_data']['sizes'] ), true));` returns for the list of image sizes:

```
array (
  0 => 'thumbnail',
  1 => 'medium',
  2 => 'full',
  3 => 'post-thumbnail',
  4 => '60x60',
  5 => '96x96',
  6 => 'third-full',
  7 => 'rect_thumb_half',
)
```

Because there's no "large" image, the thumbnail URL is `null`, and the image tag fails to load.

If we use [get_the_post_thumbnail](https://developer.wordpress.org/reference/functions/get_the_post_thumbnail/) here we get fallback to a smaller image size if the 400x300 `rect_thumb` isn't available for some reason.

Reply:
@benlk Ah, yep, I misunderstood. Updated to `get_the_post_thumbnail_url` in https://github.com/INN/umbrella-citylimits/pull/85/commits/9d308b806a85f3f09fa92484e5b7fa35d567138f